scipy/stats/tests/test_distributions.py

```diff
@@ -2969,6 +2969,93 @@ def test_entropy_transposed(self):
                                   stats.entropy(pk, qk, axis=1))
 
 
+class TestDifferentialEntropy(object):
+    """
+    Results are compared with the R package vsgoftest.
+
+    # library(vsgoftest)
+    #
+    # samp <- c(<values>)
+    # entropy.estimate(x = samp, window = <window_length>)
+
+    """
+
+    def test_differential_entropy_base(self):
+
+        random_state = np.random.RandomState(0)
+        values = random_state.standard_normal(100)
+
+        entropy = stats.differential_entropy(values)
+        assert abs(entropy - 1.342551) < 1.e-5
+
+        entropy = stats.differential_entropy(values, window_length=1)
+        assert abs(entropy - 1.122044) < 1.e-5
+
+        entropy = stats.differential_entropy(values, window_length=8)
+        assert abs(entropy - 1.349401) < 1.e-5
+
+    def test_differential_entropy_base_2d_nondefault_axis(self):
+        random_state = np.random.RandomState(0)
+        values = random_state.standard_normal((3, 100))
+
+        entropy = stats.differential_entropy(values, axis=1)
+        assert_allclose(
+            entropy,
+            [1.342551, 1.341826, 1.293775],
+            rtol=1e-6,
+        )
+
+        entropy = stats.differential_entropy(values, axis=1, window_length=1)
+        assert_allclose(
+            entropy,
+            [1.122044, 1.102944, 1.129616],
+            rtol=1e-6,
+        )
+
+        entropy = stats.differential_entropy(values, axis=1, window_length=8)
+        assert_allclose(
+            entropy,
+            [1.349401, 1.338514, 1.292332],
+            rtol=1e-6,
+        )
+
+    def test_differential_entropy_raises_value_error(self):
+        random_state = np.random.RandomState(0)
+        values = random_state.standard_normal((3, 100))
+        assert_raises(
+            ValueError,
+            stats.differential_entropy,
+            values,
+            window_length=0,
+            axis=1,
+        )
+
+        assert_raises(
+            ValueError,
+            stats.differential_entropy,
+            values,
+            window_length=50,
+            axis=1,
+        )
```

Comment:
Better here to match the error with:
```suggestion
        window_length = 0
        match_str = (f"Window length ({window_length}) must be positive and less "
                     f"than half the sample size ({len(values)}).")
        with pytest.raises(ValueError, match= match_str):
            stats.differential_entropy(
                values,
                window_length= window_length,
                axis=1,
            )
            
        window_length = 50
        match_str = (f"Window length ({window_length}) must be positive and less "
                     f"than half the sample size ({len(values)}).")
        with pytest.raises(ValueError, match= match_str):
            stats.differential_entropy(
                values,
                window_length= window_length,
                axis=1,
            )
```

Reply:
I changed the `assert_raises` to use its context manager variant. I, however, did not check for the exact string match. Is this something that you really want to test for?

Reply:
It's better to specify the error you are catching to not silently pass another error.

Reply:
I changed the tests to match the error message too.